api_server.go

```diff
@@ -30,12 +30,17 @@ var globalAPIServer = &apiServer{
 	managers: map[string]*Manager{},
 }
 
-// StartAPIServer starts the API server
-func StartAPIServer(port int) {
+// GlobalAPIHandler exports the API handler
+func GlobalAPIHandler() http.Handler {
```

Comment:
What do you think about passing a mux server here `func GlobalAPIHandler(mux *http.ServeMux) http.Handler`? That will also allow configuring the actual mux server for the globalAPI, using the same pattern as the logger.

Reply:
From what I can see, the logger is configured per-manager, which is really good. The only thing that sticks to me about the logger is that we're using the default golang logger implementation rather than an interface, meaning that we cannot easily change how go-workers2 does its logging - for example using structured logging. However, that is not relevant to this PR, it just bugs me a little but not enough to create an issue for it ;-)

For this API handler, it seems to me that it is truly global, so using the same pattern as the logger feels like overkill. I guess from my point it is about how we want to communicate that the `GlobalAPIHandler function` should be used. If we require a `http.ServeMux` we expose to callers the internals of how we actually wire up the handler, and _forever_ tie us to using `http.ServeMux`. If we simply return a `http.Handler` it leaves us free to implement routing within that handler in any way that we wish, though we may require that folks use `http.StripPrefix` to ensure that our routing logic is unaffected by any custom prefixes. Frankly the latter option to me sounds like a more flexible approach going forward.

Reply:
I am referring to these changes that just made it in master:
- https://github.com/digitalocean/go-workers2/blob/master/api_server.go#L12-L21
- https://github.com/digitalocean/go-workers2/blob/master/api_server.go#L39-L47

Reply:
This is the mux configuration options I was talking about -> https://github.com/digitalocean/go-workers2/pull/65. Your PR will be a really nice addition to this, where `func StartAPIServer(port int) {` is going to use the new func that you created but without creating a new mux, reusing the configured mux.

What do you think?

Reply:
Thank you for the context. Since that is the direction you're going to I think it makes sense to pass a `http.ServeMux` into a function that configures it with the appropriate endpoints. Let me make the change and see what you think.

Reply:
Changes made and applied to my example project: https://github.com/tomcz/example-miniredis/blob/517762c4aab3ea9a92bedc9e66be01b1219b9d3c/cmd/example/main.go#L121-L123

Custom setup is a little more verbose, but looks reasonable.